server/scsynth/SC_ComPort.cpp

```diff
@@ -278,7 +283,12 @@ class SC_UdpInPort {
             udpSocket.get_option(receiveBufferSize);
             if (receiveBufferSize.value() < SC_UdpInPort::receiveBufferSize) {
                 receiveBufferSize = SC_UdpInPort::receiveBufferSize;
-                udpSocket.set_option(receiveBufferSize);
+                boost::system::error_code ec;
+                udpSocket.set_option(receiveBufferSize, ec);
+                if (ec) {
+                    receiveBufferSize = 1024 * 1024;
+                    udpSocket.set_option(receiveBufferSize);
+                }
             }
         } catch (boost::system::system_error& e) { printf("WARNING: failed to set receive buffer size\n"); }
```

Comment:
shall we also print `e.what` or would it be redundant (I have hard time making this function fail on my system so I haven't tested it)